modules/library/render/src/main/java/org/geotools/renderer/lite/StreamingRenderer.java

```diff
@@ -2757,13 +2710,25 @@ private void drawOptimized(
             // one is there to make sure a single feature error does not ruin the rendering
             // (best effort) whilst an exception in hasNext() + ignoring catch results in
             // an infinite loop
+            boolean firstFeature = true;
+            ProjectionHandler[] handlers = new ProjectionHandler[lfts.size()];
```

Comment:
I don't understand why projection handlers and screenmaps are accumulated in arrays: they are not used after the loop, nor across iterations of the loop. Seems a simple local variable in the loop would also work.

Reply:
It is so that the projection handler and screenmaps only have to be calculated for the first feature in the loop and then they are reused for the rest of the features.  It seemed to me like this caching could be useful if there were a lot of features.

Reply:
Oh, now I see it. Yes, it makes sense. Maybe add a comment in that respect before the arrays and firstfeature? :-D

Reply:
I added a comment here.